itou/scripts/management/commands/export_auto_prescriptions.py

```diff
@@ -0,0 +1,104 @@
+from itou.siaes.enums import SiaeKind
+from itou.users.enums import KIND_SIAE_STAFF
+
+
+# this is arbitrary, it works well and gives a manageable number of objects on stdout
+# and in memory. Modify as needed if you wish so.
+DEFAULT_PAGINATION_SIZE = 1000
+
+
+def paginate_queryset(queryset, debug=False):
+    def debug_log(log):
+        if debug:
+            sys.stderr.write(log)
+
+    debug_log(f"paginated_command: queryset objects count={queryset.count()}")
+    # this is the only efficient way I found to both:
+    # - iterate over the objects and treat them on a fire-and-forget basis, without bulk updates.
+    # - iterator() and direct filter() cause OOM (out-of-memory) errors.
+    paginator = Paginator(queryset, DEFAULT_PAGINATION_SIZE)
+    try:
+        debug_log(f"paginated_command total_range={paginator.page_range}")
+        for page_number in paginator.page_range:
+            debug_log(f"paginated_command page number={page_number}")
+            page = paginator.page(page_number)
+            for obj in page.object_list:
+                yield obj
+    except KeyboardInterrupt:
+        debug_log("! keyboard interrupt, exiting.")
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument("--debug", dest="debug", action="store_true", help="Print debug statements")
+
+    def handle(self, debug=False, **kwargs):
+        writer = csv.writer(self.stdout, lineterminator=os.linesep)
+        queryset = (
+            JobApplication.objects.exclude(approval=None)
+            .select_related(
+                "approval",
+                "job_seeker",
+                "to_siae",
+                "to_siae__convention",
+            )
+            .filter(
+                state=JobApplicationWorkflow.STATE_ACCEPTED,
+                to_siae__kind__in=[SiaeKind.ACI, SiaeKind.EI, SiaeKind.ETTI, SiaeKind.AI],
+                eligibility_diagnosis__author_kind=KIND_SIAE_STAFF,
+                eligibility_diagnosis__author_siae=F("to_siae"),
+                approval__start_at__range=[datetime.date(2022, 1, 1), datetime.date(2022, 10, 1)],
+            )
+        )
+        if queryset.count() > 1:
+            writer.writerow(
+                [
+                    "ID établissement",
+                    "SIRET établissement",
+                    "SIRET à la signature",
+                    "Type établissement",
+                    "Nom établissement",
+                    "Département établissement",
+                    "Nom département établissement",
+                    "Nom région établissement",
+                    "Numéro pass",
+                    "Date début pass",
+                    "Date fin pass",
+                    "ID candidat",
+                    "Prénom candidat",
+                    "Nom candidat",
+                    "Date d'embauche",
+                ],
+            )
+
+        for ja in paginate_queryset(queryset, debug=debug):
```

Comment:
Pourquoi paginer plutôt qu’un `.iterator()` ?

Reply:
Ah ben tiens puisque tu es là je suis intéressé par la réponse. En gros j'ai essayé le `iterator()` mais je n'ai pas réussi à faire en sorte que le Ctrl+C sorte du script; avec `paginator` on sort à l'itération suivante.

Je serais très content de savoir pourquoi l'interuption ne fonctionne pas.

Reply:
Sans jeu de données, difficile d’expérimenter. Je viens de charger la base de prod, l’exécution de la commande est presque instantannée. Est-ce que tu sais comment je pourrais reproduire le problème ?

Reply:
Hm, je vais retenter le iterator, je te tiens au courant. C'etait mille fois plus simple donc je serais content que ce soit le cas.

Reply:
Ca fonctionne pour le Ctrl+C, je ne comprends pas ce qui s'etait passé.